sdk/dynamicconfig/sdk/metric/controller/push/push.go

```diff
@@ -158,12 +181,17 @@ func (c *Controller) OnUpdatedConfig(config *dynamicconfig.Config) error {
 		return err
 	}
 
+	// Clear instrument name  to collectionPeriod mappings, since
+	// they are obsolete with an updated config.
+	for name := range c.instrumentPeriod {
```

Comment:
Couldn't you just create a new map here instead of deleting every entry manually?

Reply:
Done in my new changes!